### PR TITLE
Solution

### PR DESCRIPTION
**Pull request by ArtMot99**

### CODE REVIEW

File: cinema/views.py [hunk 1]
```diff
@@ -1,45 +1,132 @@
...
-    if request.method == "DELETE":
-        movie.delete()
+    def patch(self, request, pk):
+        genre = get_object_or_404(Genre, id=pk)
+        serializer = GenreSerializer(genre, data=request.data, partial=True)
+        if serializer.is_valid():
+            serializer.save()
+            return Response(serializer.data, status=status.HTTP_200_OK)
+
+        return Response(serializer.errors, status=status.HTTP_400_BAD_REQUEST)
```
Comment (Bohdan-Salamakha) [inline cinema/views.py:66]: use is_valid(raise_exception=True) to do all checks automatically and raise handled errors if it's needed

**Review by Bohdan-Salamakha — CHANGES_REQUESTED** [review]

File: cinema/views.py [hunk 2]
```diff
@@ -1,45 +1,132 @@
...
+        serializer = GenreSerializer(genres, many=True)
+
         return Response(serializer.data, status=status.HTTP_200_OK)
 
-    if request.method == "POST":
-        serializer = MovieSerializer(data=request.data)
+    def post(self, request):
+        serializer = GenreSerializer(data=request.data)
         if serializer.is_valid():
             serializer.save()
```
Comment (EdAlekseiev) [inline cinema/views.py:36]: Use `serializer.is_valid(raise_exception=True)` instead of `if-statement`

File: cinema/views.py [hunk 3]
```diff
@@ -1,45 +1,132 @@
...
-        serializer = MovieSerializer(movie)
         return Response(serializer.data, status=status.HTTP_200_OK)
 
-    if request.method == "PUT":
-        serializer = MovieSerializer(movie, data=request.data)
+    def put(self, request, pk):
+        genre = get_object_or_404(Genre, id=pk)
+        serializer = GenreSerializer(genre, data=request.data)
         if serializer.is_valid():
             serializer.save()
```
Comment (EdAlekseiev) [inline cinema/views.py:54]: The same issue as above

**Review by EdAlekseiev — CHANGES_REQUESTED** [review]

**Review by EdAlekseiev — APPROVED** [review]
Good job

**Pull request closed without merging**